### PR TITLE
Fix #6938

### LINKED ISSUE #6938: `zig build-exe` without an argument returns an useless error message
`zig build-exe` returns:
```
error: --name [name] not provided and unable to infer
```
This is not helpful at all.

### PR DESCRIPTION
**Pull request by g-w1**
Ignore all the other commits, I accidently based this off my fork instead of ziglang/zig. Feel free to just take this code and make a commit if my commits are too messy.

### CODE REVIEW

**Comment by g-w1** [comment]
It may be useful to include `zig cc` and `zig c++` because right now it builds all the libraries when doing that even when you give it a bogus file.

**Comment by g-w1** [comment]
Never mind ignore what I just said. It builds the libraries when you give it a bogus file, but this deals with no files.

File: src/main.zig [hunk 1]
```diff
@@ -1267,9 +1267,17 @@ fn buildOutputType(
...
     {
-        fatal("`zig run` expects at least one positional argument", .{});
+        switch (arg_mode) {
+            .run => fatal("`zig run` expects at least one positional argument", .{}),
+            .build => |build_type| switch (build_type) {
+                .Exe => fatal("`zig build-exe` expects at least one positional argument", .{}),
+                .Lib => fatal("`zig build-lib` expects at least one positional argument", .{}),
+                .Obj => fatal("`zig build-obj` expects at least one positional argument", .{}),
+            },
+            else => {},
```
Comment (daurnimator) [inline src/main.zig:1279]: why not the similar message for e.g. `translate-c`

Comment (g-w1) [reply src/main.zig:1279]: It already has one:
```
$ zig translate-c
error: translate-c expects exactly 1 source file (found 0)
```
We could do it here to make it more consistent, but I think it is unnecessary.

**Pull request merged**